### PR TITLE
Add options for logit slicing to HF export

### PR DESCRIPTION
**Pull request by metascroy**
Publishes model metadata (max context length, max sequence length, vocabulary size, activation dtype, and logits-to-keep mode) as PTE constant methods and reads it back through a single typed reader in extension/llm/runner/model_metadata.h. The Python writer moves to a shared module, extension/llm/export/model_metadata.py, and the method-name strings live in one place per language (constants.h on the C++ side) so the writer and reader cannot drift apart. The reader requires the core fields and rejects malformed values; the MLX example runners use it to bound prefill, default the KV-cache storage dtype to the model's activation dtype, and check the published vocabulary size against the forward output. Exports can request full, last, or runtime-selected logits, honored by both the single-sequence and batched runners, and the C++ field prefill_chunk_size is renamed to max_seq_len to match the constant it serializes as.

### CODE REVIEW

File: backends/mlx/examples/llm/export_llm_hf.py [hunk 1]
```diff
@@ -628,6 +718,12 @@ def main():
...
     parser.add_argument("--use-offgraph-cache", action="store_true", default=False)
+    parser.add_argument(
+        "--logits-to-keep",
+        choices=("full", "last", "selected"),
+        default="full",
+        help="Logits output: full sequence, last token, or runtime-selected positions.",
+    )
     parser.add_argument(
         "--prefill-chunk-size",
         type=int,
```
Comment (kiymetakdemir) [inline backends/mlx/examples/llm/export_llm_hf.py:729]: Should --prefill-chunk-size be renamed to --max-seq-len? It bounds every forward, not just prefill and the scheduler requires a prefill chunk is smaller than what this flag sets.

Comment (metascroy) [reply backends/mlx/examples/llm/export_llm_hf.py:729]: I personally like the name prefill chunk size better, but this aligns the name with the one already used in ET (max-seq-len is what ET means by prefill chunk size; note that it is distinguished from max-context-len).

File: extension/llm/batching/module_executor.cpp [hunk 2]
```diff
@@ -271,6 +304,83 @@ Result<std::unique_ptr<ModuleExecutor>> ModuleExecutor::create(
...
+  }
+  const auto logits_info = meta->output_tensor_meta(0);
+  if (!logits_info.ok()) {
+    ET_LOG(Error, "ModuleExecutor: %s has no logits metadata", method.c_str());
+    return logits_info.error();
+  }
+  const auto logits_sizes = logits_info->sizes();
+  if (logits_sizes.size() < 2 || logits_sizes[logits_sizes.size() - 1] <= 0 ||
+      !is_supported_logits_type(logits_info->scalar_type())) {
+    ET_LOG(
```
Comment (kiymetakdemir) [inline extension/llm/batching/module_executor.cpp:362]: This check and is_supported_logits_type is defined identically here and in run_llm_hf.cpp. Could it be shared?

File: backends/mlx/examples/llm/export_llm_hf.py (hunk 1 above)
Comment (kiymetakdemir) [reply backends/mlx/examples/llm/export_llm_hf.py:729]: Thinking about single sequence it makes sense to me but my concern is about batching. Exporting with --prefill-chunk-size 1024 makes the traced width 1024, callers clamp max_batch_tokens to it, and then scheduler rejects max_prefill_chunk_size = 1024 because it needs budget larger than chunk + max_decode_sequences.

**Review by kiymetakdemir — APPROVED** [review]

File: extension/llm/batching/module_executor.cpp (hunk 2 above)
Comment (metascroy) [reply extension/llm/batching/module_executor.cpp:362]: I don't know a good place to put it

File: backends/mlx/examples/llm/export_llm_hf.py (hunk 1 above)
Comment (metascroy) [reply backends/mlx/examples/llm/export_llm_hf.py:729]: If max seq len is 1024, wouldn't we using smaller prefill chunks if we wanted to share with decode sequences?
